amd/src/package_search/mutations.js

```diff
@@ -142,4 +146,57 @@ export default class {
     async changeSort(stateManager, sort, order) {
         await this.searchPackages(stateManager, {sort: sort, order: order}, ["all", "favourites", "mine"]);
     }
+
+    /**
+     * Used to re-/load data of given categories.
+     *
+     * @param {StateManager} stateManager
+     * @param {string[]} categories
+     */
+    async load(stateManager, categories) {
+        await this.searchPackages(stateManager, {}, categories);
+    }
+
+    /**
+     * Used to un-/favourite a package.
+     *
+     * @param {StateManager} stateManager
+     * @param {int} packageid
+     * @param {boolean} favourite
+     */
+    async favourite(stateManager, packageid, favourite) {
+        const state = stateManager.state;
+        try {
+            stateManager.setReadOnly(false);
+            state.general.loading = true;
+            const successful = await favouritePackage(packageid, favourite, this.options.contextid);
+            if (!successful) {
+                return;
+            }
+            for (const category of ["all", "mine", "recentlyused"]) {
+                const pkg = state[`${category}Packages`].get(packageid);
+                if (pkg) {
+                    pkg.isfavourite = favourite;
+                }
+            }
```

Comment:
Ich denke es wäre besser, die package-Objekte weiterhin zentral in einer Liste im State zu speichern, sodass du hier nicht über alle Kategorien iterieren musst (und der State keine redundanten Daten enthält). In `state[`${category}Packages\`]` könntest du dann z.B. nur eine Liste der package-IDs speichern. Beim rendern des Tab-Inhaltes holst du dir dann die Daten des Packages aus der zentralen Liste. Die Package-Komponente muss dann nicht wissen, in welcher Kategorie sie angezeigt wird.